sys/net/include/netmod.h

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (C) 2015 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    net_netmod Global network module list
+ * @ingroup     net
+ * @brief       Global list of available network module
+ * @{
+ *
+ * @file
+ * @brief       List of all available network modules
+ *
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ */
+
+#ifndef __NETMOD_H_
+#define __NETMOD_H_
+
+
+typedef enum {
```

Comment:
Would start with something like `NETMOD_UNDEF` for unidentified data. To save space in the lookup-array and to have the correct behaviour for `NETMOD_NUMOF` I would suggest:

```C
typedef enum __attribute__((packed)) { // packed to make enum as small as possible
    NETMOD_UNDEF = -1,
#ifdef MODULE_UDP
    NETMOD_UDP,
#endif
/* … */
    NETMOD_NUMOF
}